osquery/tables/ipmi/ipmi_info.cpp

```diff
@@ -0,0 +1,1309 @@
+    LOG(ERROR) << "Could not get ipmi_lanparm_t for mc '" << key;
+    return;
+  }
+
+  char name[IPMI_MC_NAME_LEN];
+  int len = ipmi_mc_get_name(mc, name, IPMI_MC_NAME_LEN);
+  if (len > 0) {
+    client->insertRowsQueue(key, "mc_name", name);
+  }
+  client->insertRowsQueue(key, "mc_id", std::to_string(ipmi_mc_device_id(mc)));
+
+  getPARM(lp, IPMI_LANPARM_IP_ADDRESS, key, "ip", client);
+
+  getPARM(lp, IPMI_LANPARM_DEFAULT_GATEWAY_ADDR, key, "gateway_ip", client);
+
+  getPARM(lp, IPMI_LANPARM_MAC_ADDRESS, key, "mac", client);
+
+  getPARM(lp, IPMI_LANPARM_SUBNET_MASK, key, "subnet_mask", client);
+
+  getPARM(
+      lp, IPMI_LANPARM_DEFAULT_GATEWAY_MAC_ADDR, key, "gateway_mac", client);
+
+  getPARM(
+      lp, IPMI_LANPARM_BACKUP_GATEWAY_ADDR, key, "backup_gateway_ip", client);
+
+  getPARM(lp,
+          IPMI_LANPARM_BACKUP_GATEWAY_MAC_ADDR,
+          key,
+          "backup_gateway_mac",
+          client);
+
+  getPARM(lp, IPMI_LANPARM_IP_ADDRESS_SRC, key, "ip_address_source", client);
+
+  getPARM(
+      lp, IPMI_LANPARM_COMMUNITY_STRING, key, "snmp_community_string", client);
+
+  getPARM(lp, IPMI_LANPARM_SET_IN_PROGRESS, key, "set_in_progress", client);
+}
+
+void IPMIFullyUpCB(ipmi_domain_t* domain, void* data) {
+  TLOG << "OpenIPMI is now fully up";
+  IPMIClient* c = (IPMIClient*)data;
+  c->setDomain(domain);
+
+  return;
+}
+
+/// Gets the the value suffix for a sensor reading.
+std::string getSensorThresholdSuffix(ipmi_sensor_t* sensor) {
+  const size_t suffixLen = 50;
+  char suffix[suffixLen];
+
+  const char* percent = "";
+  const char* base;
+  const char* mod_use = "";
+  const char* modifier = "";
+  const char* rate;
+  base = ipmi_sensor_get_base_unit_string(sensor);
+  if (ipmi_sensor_get_percentage(sensor))
+    percent = "%";
+  switch (ipmi_sensor_get_modifier_unit_use(sensor)) {
+  case IPMI_MODIFIER_UNIT_NONE:
+    break;
+  case IPMI_MODIFIER_UNIT_BASE_DIV_MOD:
+    mod_use = "/";
+    modifier = ipmi_sensor_get_modifier_unit_string(sensor);
+    break;
+  case IPMI_MODIFIER_UNIT_BASE_MULT_MOD:
+    mod_use = "*";
+    modifier = ipmi_sensor_get_modifier_unit_string(sensor);
+    break;
+  }
+  rate = ipmi_sensor_get_rate_unit_string(sensor);
+
+  snprintf(
+      suffix, suffixLen, "%s %s%s%s%s", percent, base, mod_use, modifier, rate);
+
+  return suffix;
+}
+
+void readThresholdSensorCB(ipmi_sensor_t* sensor,
+                           int err,
+                           enum ipmi_value_present_e value_present,
+                           unsigned int raw_value,
+                           double val,
+                           ipmi_states_t* states,
+                           void* data) {
+  if (err != 0) {
+    LOG(ERROR) << "Could not read sensor: " << strerror(err);
+    return;
+  }
+
+  Row r;
+
+  ipmi_mc_t* mc = ipmi_sensor_get_mc(sensor);
+  r["mc_name"] = getMCName(mc, "unknown");
+  r["mc_id"] = INTEGER(ipmi_mc_device_id(mc));
+
+  const int maxChar = 256;
+  char name[maxChar];
```

Comment:
nit: There's quite a few of these instances of initializing a `char` array. Perhaps these would be better handled by something like a `std::vector<char>` and initializing the values to `0`?